jobs/AnalyzeResults.php

```diff
@@ -0,0 +1,136 @@
+            $precisionTop30s[$resultset['term']] = $this->calculatePrecisionTop30(
+                $resultset['id'], $resultset['term'] );
+            fwrite( $this->out,
+                $resultset['term'] . "," .
+                $f1Scores[$resultset['term']] . "," .
+                $precisionTop30s[$resultset['term']] . "\n" );
+        }
+        fwrite(
+            $this->out,
+            "ARITHMETIC MEAN," .
+            array_sum( $f1Scores ) / count( $resultsets) . "," .
+            array_sum( $precisionTop30s ) / count( $resultsets) .
+            "\n"
+        );
+    }
+
+    private function calculateF1Score( int $resultsetId, string $searchTerm ) : float {
+        $knownGoodImageCount = $this->db->query(
+            'select count(*) as count from results_by_component where ' .
+            'term = "' . $this->db->real_escape_string( trim( $searchTerm ) ) . '" and ' .
+            'rating=1'
+        )->fetch_object()->count;
+        $foundGoodImageCount = $this->db->query(
+            'select count(*) as count from labeledResult where ' .
+            'resultsetId = "' . intval( $resultsetId ) . '" and ' .
+            'rating=1'
+        )->fetch_object()->count;
+        if ( $foundGoodImageCount == 0 ) {
+            return 0;
+        }
+        $foundBadImageCount = $this->db->query(
+            'select count(*) as count from labeledResult where ' .
+            'resultsetId = "' . intval( $resultsetId ) . '" and ' .
+            'rating<1'
+        )->fetch_object()->count;
+
+        $precision = $foundGoodImageCount / ( $foundGoodImageCount + $foundBadImageCount );
+        $recall = $foundGoodImageCount / $knownGoodImageCount;
+
+        return 2 * ( ( $precision * $recall ) / ( $precision + $recall ) );
+    }
+
+    /**
+     * @param int $resultsetId
+     * @param string $searchTerm
+     * @return float
+     */
+    private function calculatePrecisionTop30( int $resultsetId, string $searchTerm ) : float {
+        $knownGoodImageCount = $this->db->query(
+            'select count(*) as count from results_by_component where ' .
+            'term = "' . $this->db->real_escape_string( trim( $searchTerm ) ) . '" and ' .
+            'rating=1 and position < 31'
+        )->fetch_object()->count;
+        $foundGoodImageCount = $this->db->query(
+            'select count(*) as count from labeledResult where ' .
+            'resultsetId = "' . intval( $resultsetId ) . '" and ' .
+            'rating=1'
+        )->fetch_object()->count;
+        if ( $foundGoodImageCount == 0 ) {
+            return 0;
+        }
+        $foundBadImageCount = $this->db->query(
+            'select count(*) as count from labeledResult where ' .
+            'resultsetId = "' . intval( $resultsetId ) . '" and ' .
+            'rating<1 and position < 31'
+        )->fetch_object()->count;
+
+        $precision = $foundGoodImageCount / ( $foundGoodImageCount + $foundBadImageCount );
```

Comment:
When both foundGoodImageCount === 0 && foundBadImageCount === 0, we should ignore this value completely. We shouldn't treat it as not having found a good value - we simply don't know what it found. All results might be perfect, we just don't have any labeled sample in there to inform us.